token/token.go

```diff
@@ -196,7 +202,7 @@ const LowestPrec = 0
 // Precedence returns the precedence for the operator token.
 func (tok Token) Precedence() int {
 	switch tok {
-	case LOr:
+	case LOr, NullCoalesce:
```

Comment:
`NullCoalesce` operator should have the lowest precedence as 1, and we should increment the others accordingly. It creates ambiguity if we let it be the same as `LOr`. AFAIK, JS also does not permit the use of `??` with other operators without parentheses.